### PR TITLE
{FEC} Application support

### PR DESCRIPTION
**Pull request by ethouris**
Changes:

* New `packetfilter` option supported
* Updated statistics for packet filter
* Added support for json format also for testing apps

### CODE REVIEW

**Comment by ethouris** [comment]
BLOCKS: 858

File: srtcore/packetfilter.cpp [hunk 1]
```diff
@@ -232,7 +232,9 @@ void PacketFilter::globalInit()
     // as builtin. This will disallow users to register
     // external filters with the same name.
 
-    filters["fec"] = new Creator<FECFilterBuiltin>;
+
+    ManagedPtr c = new Creator<FECFilterBuiltin>;
+    filters["fec"] = c;
```
Comment (maxlovic) [inline srtcore/packetfilter.cpp]: What is the purpose?

Comment (ethouris) [reply srtcore/packetfilter.cpp]: That was during the development I had a problem with the value of `ManagedPtr` in this operation. Now probably redundant.

Comment (maxlovic) [reply srtcore/packetfilter.cpp]: Please delete.

Comment (ethouris) [reply srtcore/packetfilter.cpp]: Well, I changed my mind :D

**Pull request merged**